test.py

```diff
@@ -330,21 +312,7 @@ async def reap(done, pending, signaled):
                 console.print_progress(result)
         return failed
 
-    ms = MinioServer(options.tmpdir, '127.0.0.1', LogPrefixAdapter(logging.getLogger('minio'), {'prefix': 'minio'}))
-    await ms.start()
-    TestSuite.artifacts.add_exit_artifact(None, ms.stop)
-
-    hosts = HostRegistry()
-    mock_s3_server = MockS3Server(await hosts.lease_host(), 2012,
-                                  LogPrefixAdapter(logging.getLogger('s3_mock'), {'prefix': 's3_mock'}))
-    await mock_s3_server.start()
-    TestSuite.artifacts.add_exit_artifact(None, mock_s3_server.stop)
-
-    minio_uri = "http://" + os.environ[ms.ENV_ADDRESS] + ":" + os.environ[ms.ENV_PORT]
-    proxy_s3_server = S3ProxyServer(await hosts.lease_host(), 9002, minio_uri, 3, int(time.time()),
-                                    LogPrefixAdapter(logging.getLogger('s3_proxy'), {'prefix': 's3_proxy'}))
-    await proxy_s3_server.start()
-    TestSuite.artifacts.add_exit_artifact(None, proxy_s3_server.stop)
+    await start_s3_mock(minio_tempdir_base=options.tmpdir)
```

Comment:
nit. I would rename it. It's starting the Minio, S3 proxy and S3 mock.

Reply:
Any suggestions?

Reply:
`prepare_environment` or `start_3rd_party_services`?

Reply:
changed to `start_s3_mock_services`

Reply:
Well, currently that's fine, but we both already know that there will be toxiproxy for LDAP as well :)